src/reconcile/rec_visibility.c

```diff
@@ -248,6 +248,75 @@ __timestamp_no_ts_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw)
     return (false);
 }
 
+/*
+ * __rec_delete_hs_upd --
+ *     Delete the value left in the history store
+ */
+static int
+__rec_delete_hs_upd(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ROW *rip, uint64_t recno,
+  WT_UPDATE *delete_upd, WT_UPDATE *delete_tombstone)
+{
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_ITEM hs_recno_key;
+    uint8_t *p, hs_recno_key_buf[WT_INTPACK64_MAXSIZE];
+    bool hs_read_committed_set;
+
+#ifdef HAVE_DIAGNOSTIC
+    WT_TIME_WINDOW *hs_tw;
+#endif
+
+    /* Open a history store cursor if we don't yet have one. */
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+    hs_read_committed_set = F_ISSET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    if (r->page->type == WT_PAGE_ROW_LEAF) {
```

Comment:
Is there a pattern spread across our code of building a history store key? It seems complex, and layers in knowledge about row/column stores which it's better to keep out of general reconciliation code like this (i.e: I'd rather have fewer places that conditionally check for row vs column store pages.

Reply:
There are other places we do the same. I can extract it to a separate function.

Reply:
This is not needed anymore after I moved the code to hs_rec.c